Automattic/simplenote-ios
# Fixing Markdown Gestures

### Fix
In this PR we're fixing a bug that caused text selection in the Markdown UI to accidentally dismiss the topViewController.

@aerych May you do the honors? **Thanks in advance!!!**

Closes #435

### Details:
We're letting `UINavigationController.interactivePopGestureRecognizer` deal with **pop** gestures performed over the MarkdownPreview UI.

- `UINavigationController.interactivePopGestureRecognizer.delegate` must be implemented, since having a custom TitleBar in the Notes Editor disables its standard behavior
- We're keeping a close eye on #490 since that's why such delegate was removed, in the first place
- Ref. this [SO Issue](https://stackoverflow.com/questions/34942571/how-to-enable-back-left-swipe-gesture-in-uinavigationcontroller-after-setting-le?noredirect=1&lq=1)

### Test
1. Launch Simplenote and log into your account
2. Open (any) Markdown Enabled note
3. Swipe from he center of the screen towards the left edge, in order to reveal Markdown Preview
4. Verify you can select text, and extend the selection from both ends
5. Verify you can swipe to dismiss the Markdown Preview UI from the left edge
6. Verify you can swipe to dismiss the Editor's UI


**Sample video below!!**

### Release
`RELEASE-NOTES.txt` was updated in 95ebede with:

> Fixed a bug that caused the Markdown Preview UI to get dismissed accidentally


---

### Demo:

![Demo](https://user-images.githubusercontent.com/1195260/68161114-e90eff80-ff33-11e9-9f12-d5fa8499a291.gif)
